crates/ruff/tests/format.rs

```diff
@@ -90,6 +90,172 @@ fn format_warn_stdin_filename_with_files() {
     "###);
 }
 
+#[test]
+fn nonexistent_config_file() {
+    assert_cmd_snapshot!(Command::new(get_cargo_bin(BIN_NAME))
+        .args(["format", "--config", "foo.toml", "."]), @r###"
+        success: false
+        exit_code: 2
+        ----- stdout -----
+
+        ----- stderr -----
+        error: invalid value 'foo.toml' for '--config <CONFIG_OPTION>'
+
+          tip: The `--config` flag must either be a path to a `.toml` configuration file or a TOML `<KEY> = <VALUE>` pair overriding a specific config setting
+          tip: The path `foo.toml` does not exist on your filesystem
+
+        For more information, try '--help'.
+       "###);
+}
+
+#[test]
+fn config_override_rejected_if_invalid_toml() {
+    assert_cmd_snapshot!(Command::new(get_cargo_bin(BIN_NAME))
+        .args(["format", "--config", "foo = bar", "."]), @r###"
+    success: false
+    exit_code: 2
+    ----- stdout -----
+
+    ----- stderr -----
+    error: invalid value 'foo = bar' for '--config <CONFIG_OPTION>'
+
+      tip: The `--config` flag must either be a path to a `.toml` configuration file or a TOML `<KEY> = <VALUE>` pair overriding a specific config setting
+      tip: The following error occurred when attempting to parse `foo = bar` as a `ruff.toml` config option:
+
+    TOML parse error at line 1, column 7
+      |
+    1 | foo = bar
+      |       ^
+    invalid string
+    expected `"`, `'`
+
+    For more information, try '--help'.
+    "###);
+}
+
+#[test]
+fn too_many_config_files() -> Result<()> {
+    let tempdir = TempDir::new()?;
+    let ruff_dot_toml = tempdir.path().join("ruff.toml");
+    let ruff2_dot_toml = tempdir.path().join("ruff2.toml");
+    fs::File::create(&ruff_dot_toml)?;
+    fs::File::create(&ruff2_dot_toml)?;
+    let expected_stderr = format!(
+        "\
+ruff failed
+  Cause: You cannot specify more than one configuration file on the command line.
+
+  tip: remove either `--config={}` or `--config={}`.
```

Comment:
This is a confusing message. Why do we have two `{}`? Shouldn't these have file paths?

Reply:
The message the user sees will have file paths:

![image](https://github.com/astral-sh/ruff/assets/66076021/596533bf-f91a-4c71-acd3-897171711af4)

This test doesn't use `assert_cmd_snapshot`, because we don't know what the file paths are going to be ahead of time in this test (it uses a TempDir), and `assert_cmd_snapshot` needs a literal string to be passed in. So this is a format string that I'm using to generate the expected stderr given the path to the TempDir, and then I'm just using an `assert_eq` call lower down once I've generated the expected stderr.

Reply:
Oh. I would use insta filters for this e.g. https://github.com/astral-sh/ruff/pull/9764/commits/d467bc6344b7b83546b24a4f6b576e76497afdfe (although it's kind of a pain for Windows https://github.com/astral-sh/ruff/pull/9764/commits/f46c2e4edc7861024a690c8324b07a97d5d346af)

Reply:
Ahh makes sense, thanks! I looked for something like that in the insta_cmd docs, but obviously I didn't look thoroughly enough 😄

Reply:
Fixed in https://github.com/astral-sh/ruff/pull/9599/commits/7739440e40eaebab41b3fed68a0e25b9b458777e